### ISSUE TITLE
vscode-eslint unable to resolve path to module

### ISSUE THREAD

**Issue by creplogle** [microsoft/vscode-eslint#960]
I have a React project that used .js extensions but to comply with coding standards components need to use .jsx extensions. When I changed src/components/Root.js to src/components/Root.jsx vscode-eslint displays an "Unable to resolve path to module ./components/Root. No error with .js, error with .jsx. Running eslint from the command line does not emit any errors. I've seen the closed issues related to this but don't understand why I need special configuration for VS Code to work with a correct project.

**Comment by dbaeumer** [comment]
When you run eslint from the terminal do you change into a special directory. If this is the case VS Code needs to know about this directory otherwise it can't use the same module lookup rules.

Can you please provide me with a GitHub repository I can clone that demos what you are experiencing.

**Comment by creplogle** [comment]
I'm running `npm run lint` from my project root, the same root that is used for the VS Code workspace. The project was generated from this template https://github.com/coryhouse/react-slingshot, then refactored for the actual app that is in a private repo. Cory's template uses .js extension, but to comply with coding standards I refactored index.js, Root.js, App.js, etc. to use .jsx extensions. When I loaded the initial generated app vscode-elint did not complain about things like

`import Root from './components/Root';`

in index.js. Nor did it complain when I changed index.js to index.jsx. But when I refactored src/components/Root.js to src/components/Root.jsx it complain when running

`npm run lint`

in my project root does not report errors. Again I'm running `npm run lint` from the project root directory (src is a direct subdir), and the project root is the project root for VS Code.

**Comment by creplogle** [comment]
I'll create a sample this weekend, trying to meet project deadlines at the moment.

**Comment by dbaeumer** [comment]
Great, ping me when available.

**Comment by creplogle** [comment]
Dirk, thanks for your patience. Here is a sample repo that recreates the issue. Clone the repo and load into VS Code on the master branch and you won't see any eslint errors. Change to the issue-960 branch and you will see an eslint error when you open src/index.jsx (line 7). If you run

`npm run lint`

in the console eslint does not report any errors, and 

`npm start -s`

runs the app with no errors.

**Comment by creplogle** [comment]
It would probably be helpful if I included the repo link...

[https://github.com/creplogle/eslint-960](url)

**Comment by dbaeumer** [comment]
Actually this is a configuration isssue with ESLint itself. If executed in the terminal I get:

```
dirkb:~/Projects/../eslint-960 [issue-960|✔] $ ./node_modules/.bin/eslint src/index.jsx
Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration .

/home/dirkb/Projects/mseng/VSCode/Playgrounds/eslint-960/src/index.jsx
  7:18  error  Unable to resolve path to module './components/Root'  import/no-unresolved

✖ 1 problem (1 error, 0 warnings)

dirkb:~/Projects/../eslint-960 [issue-960|✔] $ 
```

Since the plugin simply calls into ESLint itself it will surface the same problems. So all I can recommend is to get ESLint running correctly in the terminal. If this is the case the ESLint extension should work as well.

**Comment by dbaeumer** [comment]
Please ping if you thing there is anything the extension can do.

**Comment by creplogle** [comment]
Thanks for looking into this, sorry for your trouble.
